app/services/nomis/elite2/offender_api.rb

```diff
@@ -30,23 +30,44 @@ def self.list(prison, page = 0, page_size: 20)
           [data, total_pages]
         }
 
+        search_route = '/prisoner-numbers'
+        offenders = data.map { |o| o.fetch('offenderNo') }
+        search_key = "#{search_route}_#{Digest::SHA256.hexdigest(offenders.to_s)}"
+        search_response = Rails.cache.fetch(search_key,
+                                            expires_in: Rails.configuration.cache_expiry) {
+                            search_client.post(search_route, prisonerNumbers: offenders)
+                          }.index_by { |o| o.fetch('prisonerNumber') }
+
+        offender_list = data.map do |offender|
+          recall_data = search_response.fetch(offender.fetch('offenderNo'), {})
+          offender.merge('recall' => recall_data.fetch('recall', false))
+        end
+
         offenders = api_deserialiser.deserialise_many(
-          Nomis::OffenderSummary, data)
+          Nomis::OffenderSummary, offender_list)
         ApiPaginatedResponse.new(total_pages, offenders)
       end
 
       def self.get_offender(offender_no)
         # Bad NOMIS numbers mustn't produce invalid URLs
-        route = "/prisoners/#{URI.encode_www_form_component(offender_no)}"
-        response = Rails.cache.fetch("offender-#{route}",
+        url_offender_no = URI.encode_www_form_component(offender_no)
+        route = "/prisoners/#{url_offender_no}"
+        response = Rails.cache.fetch(route,
                                      expires_in: Rails.configuration.cache_expiry) {
           e2_client.get(route)
         }
 
         if response.empty?
           nil
         else
-          api_deserialiser.deserialise(Nomis::Offender, response.first)
+          recall_route = "/offenders/#{url_offender_no}"
+          recall_response = Rails.cache.fetch(recall_route,
+                                              expires_in: Rails.configuration.cache_expiry) {
+            e2_client.get(recall_route)
+          }
+          # api_deserialiser.deserialise(Nomis::Offender, response.first)
+          api_deserialiser.deserialise(Nomis::Offender,
+                                       response.first.merge('recall' => recall_response.fetch('recall', false)))
```

Comment:
One thing to note with this implementation is that we have two different 'sources of truth' for the recall field, depending on how you get offender(s):

- `.list` – get all offenders in a prison – uses the Prison Offender Search API
- `.get_offender` – get a single offender – uses the Prison API

Are we sure it's safe and sensible to make the assumption that the two APIs will be consistent in the recall status that they give? Can we treat them both as a 'source of truth'?

I'd be tempted to say they should both be backed by the Prison Offender Search API. Given the fact that we need to perform an additional API request anyway, there shouldn't be much difference between going to the Prison Offender Search API instead of the Prison API to get the recall status.